compiler/rustc_smir/src/stable_mir/mir/body.rs

```diff
@@ -52,6 +53,52 @@ pub enum UnwindAction {
     Cleanup(usize),
 }
 
+#[derive(Clone, Debug)]
+pub enum AssertMessage {
```

Comment:
Do you think we can also add a method to get the message description?

Reply:
@oli-obk I've added a naive impl of description in https://github.com/rust-lang/rust/pull/110610/commits/06d872de01e4f6b3302870be8c4363b844f7d312. I guess we don't want to duplicate things in this way?. Unsure if we've added impls of things that we implement in MIR but now in SMIR and how we did so. The other possibility is that `description` in SMIR just gets smir::Assert and get mir::Assert and call its description method.

Reply:
You could just grab the description from the MIR and store it in a field on the SMIR to avoid duplication at the cost of some memory.

What is the purpose of having this description available for SMIR?

Reply:
We can't lazily get the MIR object from a SMIR one, there's no connection after the conversion